core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -935,7 +935,7 @@ private Builder(TableMetadata base) {
       this.base = base;
       this.formatVersion = base.formatVersion;
       this.uuid = base.uuid;
-      this.lastUpdatedMillis = null;
+      this.lastUpdatedMillis = base.lastUpdatedMillis;
       this.location = base.location;
```

Comment:
I am looking at the [spec](https://iceberg.apache.org/spec/#table-metadata-fields)

>  Each table metadata file should update this field just before writing.

I believe this shall be set on write instead of in builder initialization. 
- If there's new snapshot in table metadata, we will update this to be snapshot timestamp. 
- [Else if empty table metadata, it will be set to current time when build]( https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1433-L1435)